src/mrpro/algorithms/optimizers/pgd.py

```diff
@@ -0,0 +1,158 @@
+"""Proximal Gradient Descent algorithm."""
+
+import math
+from collections.abc import Callable
+
+import torch
+from typing_extensions import Unpack
+
+from mrpro.algorithms.optimizers.OptimizerStatus import OptimizerStatus
+from mrpro.operators import ProximableFunctionalSeparableSum
+from mrpro.operators.Functional import ProximableFunctional
+from mrpro.operators.Operator import Operator
+
+
+class PGDStatus(OptimizerStatus):
+    """Proximal Gradient Descent callback base class."""
+
+    stepsize: float
+    """Current stepsize of the algorithm."""
+
+
+def pgd(
+    f: Operator[torch.Tensor, tuple[torch.Tensor]] | Operator[Unpack[tuple[torch.Tensor, ...]], tuple[torch.Tensor]],
+    g: ProximableFunctional | ProximableFunctionalSeparableSum,
+    initial_value: torch.Tensor | tuple[torch.Tensor, ...],
+    stepsize: float = 1.0,
+    reg_parameter: float = 0.01,
+    max_iterations: int = 128,
+    backtrack_factor: float = 1.0,
+    callback: Callable[[PGDStatus], None] | None = None,
```

Comment:
since it is probably quite straight forward to implement, we could also think of having both ISTA and FISTA with this code? We would need some sort of flag to distinguish the two cases, though. I could imagine that @fzimmermann89 will have an opinion about how nice this is or not? ;)